server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -211,6 +241,11 @@ public DesiredBalance compute(
             }
         }
 
+        final int iterationCountWarningInterval = computeIterationCountWarningInterval(routingAllocation);
+        final long timeWarningInterval = progressLogInterval.millis();
+        final long computationStartedTime = threadPool.rawRelativeTimeInMillis();
+        long nextWarningTime = computationStartedTime + timeWarningInterval;
```

Comment:
Also, naming nit (my fault): we're not _warning_ so I think we should rename these things `nextReportTime`, `shouldReportByTime`, etc.